internal/ad/backends/winbind/winbind.go

```diff
@@ -109,9 +126,24 @@ func (w Winbind) Domain() string {
 	return w.domain
 }
 
-// HostKrb5CCNAME returns the absolute path of the machine krb5 ticket.
-func (w Winbind) HostKrb5CCNAME() string {
-	return w.hostKrb5CCNAME
+// HostKrb5CCName returns the absolute path of the machine krb5 ticket.
+func (w Winbind) HostKrb5CCName() (string, error) {
+	target := "/tmp/krb5cc_0"
+
+	// Uppercase domain and hostname
+	domain := strings.ToUpper(w.domain)
+	hostname := strings.ToUpper(w.hostname)
+
+	principal := fmt.Sprintf("%s$@%s", hostname, domain)
+	cmdArgs := append(w.kinitCmd, "-k", principal, "-c", target)
```

Comment:
As we are using kinit, shouldn’t the adsys package depends on klibc-utils explicitly?